proof/dleq/dleq.go

```diff
@@ -91,20 +117,32 @@ func NewDLEQProofBatch(suite Suite, G []kyber.Point, H []kyber.Point, secrets []
 	}
 
 	// Collective challenge
-	h := suite.Hash()
+	hSuite := suite.Hash()
 	for _, x := range xG {
-		x.MarshalTo(h)
+		_, err := x.MarshalTo(hSuite)
+		if err != nil {
+			return nil, nil, nil, err
+		}
 	}
 	for _, x := range xH {
-		x.MarshalTo(h)
+		_, err := x.MarshalTo(hSuite)
+		if err != nil {
+			return nil, nil, nil, err
+		}
 	}
 	for _, x := range vG {
-		x.MarshalTo(h)
+		_, err := x.MarshalTo(hSuite)
+		if err != nil {
+			return nil, nil, nil, err
+		}
 	}
 	for _, x := range vH {
-		x.MarshalTo(h)
+		_, err := x.MarshalTo(hSuite)
+		if err != nil {
```

Comment:
This kind of cases could be written as:

```suggestion
		if _, err := x.MarshalTo(hSuite); err != nil {
```